scripts/Jenkinsfiles/quality

```diff
@@ -0,0 +1,219 @@
+def runQualityTests() {
+    sshagent(credentials: ['jenkins-worker'], ignoreMissing: true) {
+        // Determine git refspec, branch, and clone type
+        git_shallow_clone = true
+        git_extensions = []
+        if (env.ghprbActualCommit) {
+            git_branch = "${ghprbActualCommit}"
+            refspec = "+refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/*"
```

Comment:
What's going on here with SHARD 4?

Reply:
Shard 4 needs master, and a higher depth than 1 in order to run the xss commit linter. Other stages are working off a shallow clone of the commit's branch. Rather than increase timings of the checkout for every stage, I extend the refspec and replace the shallow clone.

Reply:
ah, understood. Thanks